components/list/list-item-drag-mixin.js

```diff
@@ -98,3 +99,84 @@ export const ListItemDragMixin = superclass => class extends superclass {
 		` : nothing;
 	}
 };
+
+export class NewPositionEventDetails {
+	/**
+	 * @param { Object } object An simple object with the position event properties
+	 * @param { String } object.targetKey The item key of the list-item that is moving
+	 * @param { String } object.destinationKey The item key of the list-item in the position we are moving to
+	 * @param { String } object.temporaryMovement Information on whether the item is entering or exiting temporary movement
+	 */
+	constructor(object) {
+		this.targetKey = object.targetKey;
+		this.destinationKey = object.destinationKey;
+		this.temporaryMovement = object.temporaryMovement;
+	}
+
+	/**
+	 * Announces a move to screen readers on an array
+	 * @param { Array<Node> } list The array to announce the move on.
+	 * @param { function(any, Number): String } announceCallback Callback function that returns the announcement text
+	 *        Takes the item in the array and the index as arguments. Should return the text to announce.
+	 */
+	announceMove(list, announceCallback) {
+		const index = this.fetchPosition(list, this.targetKey);
+		if (index === null) throw new Error('Item to announce move not found in given array.');
+		const item = list[index];
+		const message = announceCallback(item, index);
+		if (message) announce(message);
+	}
+
+	/**
+	 * Fetches an index position within a list given an item key
+	 * @param { Array<Node> } list Array of nodes
+	 * @param { String } key Item key to fetch position of
+	 */
+	fetchPosition(list, key) {
```

Comment:
Let key be a callback so a user can define their own key. Not all arrays will have `.key` property. We might want to use the `href` or `self` as a key.

Reply:
could key property name be passed via constructor ?

Reply:
Callback works better I think - it's consistent with what `lit-html` does and is more versatile than a property name

Reply:
The list is constructing and not the user. So it won't know about where the key is either.